cmd/root.go

```diff
@@ -146,6 +146,11 @@ func (authArgs *authArgs) validateAuthArgs() error {
 		return errors.New("--auth-method is a required parameter")
 	}
 
+	// Back-compat to accommodate existing client usage patterns that assume that "client-secret" is the default
```

Comment:
@jadarsie Do you have any concerns here for Azure Stack? Does the `az` CLI auth model work for Azure Stack? If not, are you comfortable with this back-compat solution that "re-sets" the default auth method to `client_secret` if the service principal ID and password are included in the command line arguments?

The goal for us is to default to CLI as the auth model because for most users it is easier (don't have to generate/maintain service principals, easier command statements). If there's a reason that a local `az` context isn't possible, then this back-compat solution should work for pre-existing scripts that pass in the id and pass but haven't ever bothered to set the `--auth-method` to `client_secret` explicitly, because it's always been the default.

Reply:
ping @jadarsie @haofan-ms